eyeshade/controllers/publishers.js

```diff
@@ -109,12 +109,9 @@ v2.settlement = {
         publisher = await publishers.findOne({ publisher: entry.publisher })
         if (!publisher) return reply(boom.badData('no such entry: ' + entry.publisher))
 
-        if (!publisher.owner) return reply(boom.badData('no owner for publisher: ' + entry.publisher))
-
-        owner = await owners.findOne({ owner: publisher.owner })
+        // The owner at the time of uploading could be different
```

Comment:
right, but why isn't the previous owner getting the settlement? if we're going to fix this, we really ought to take each vote with the owner at the time the vote is recorded.  so, if owner X sells publisher Z to owner Y on such-and-such a date, then votes cast upto that time, should go to owner X, not owner Y, irregardless of when the settlement report is generated.

the change above is probably no more wrong than the existing code, but it is not fixing the actual problem...

Reply:
the owner in the settlement file is the one that was actually paid - this change made it so the settlement entry in db always reflected that